text/0006-shipyard-next-generation.md

```diff
@@ -0,0 +1,317 @@
+# The next generation of Shipyard
+
+The next generation of Shipyard to allow custom delivery/remediation processes.
+
+## Motivation
+
+*There's always room for smart enhancements* - the Shipyard specification is no exception.
+
+A shipyard allows a **Site Reliability Engineer (SRE)** to specify WHAT should happen in the course of an artifact delivery or remediation process. Due to certain use cases, it has been identified that the current specification of the shipyard has limitations and hides details from the SRE. This KEP proposes an enhancement of the *Shipyard* specification to bring flexibility into the definition of the *WHAT*, but still provides an opinionated approach and smart defaults.
+
+Please read this KEP in combination with the KEP: [First version of a Keptn uniform](https://github.com/keptn/enhancement-proposals/pull/7); especially regarding the eventing part.
+
+## Explanation
+
+:man:/:blonde_woman: The target persona of this KEP is an *SRE*, who is responsible for processes and workflows to automate application delivery or remediation operations. 
+
+Before getting started, a brief recap of the important terms and definitions used in this KEP:
+
+- **Project stage:** A project stage (or just stage) defines a logical space in an environment, which has a dedicated purpose for an application in a continuous delivery process. The project stages are ordered, meaning that a stage has at least a previous and/or next stage. 
+
+- **Shipyard:** A shipyard is the declarative means to divide an environment into project stages and to specify task sequences for each stage.
+
+- **Sequence:** A sequence declares a set of tasks, which are triggered by an external event (aka. domain event, e.g.: artifact built → deploy artifact, a problem occurred → execute remediation, etc.). A sequence executes the tasks in the given order. 
+
+- **Task:** A task is the smallest executable unit of a sequence. A task is kicked-off by a *triggered* event and has a life cycle of a *start* and *finished* event. 
```

Comment:
*start*, *status.changed*, and *finished* event.

Reply:
*start* --> *started*
I forgot about the *status.changed* which is optional. 

I updated the KEP accordingly.